lib/spack/spack/build_systems/cached_cmake.py

```diff
@@ -248,12 +260,22 @@ def initconfig_hardware_entries(self):
             # Include the deprecated CUDA_TOOLKIT_ROOT_DIR for supporting BLT packages
             entries.append(cmake_cache_path("CUDA_TOOLKIT_ROOT_DIR", cudatoolkitdir))
 
+            cuda_flags = []
             archs = spec.variants["cuda_arch"].value
             if archs[0] != "none":
                 arch_str = ";".join(archs)
                 entries.append(
                     cmake_cache_string("CMAKE_CUDA_ARCHITECTURES", "{0}".format(arch_str))
                 )
+                # TODO: Additional definitions that may not be needed anymore
+                # This is an imperfect merge of the radiuss way and the "spack"
+                # way: radiuss used to define only the first arch in the list,
+                # even for CMAKE_CUDA_ARCHITECTURE. What do we want?
+                cuda_flags.append("-arch sm_{0}".format(archs[0]))
+                entries.append(cmake_cache_string("CUDA_ARCH", "sm_{0}".format(archs[0])))
```

Comment:
We want the list.  On our machines where we usually do separate builds for each one, it doesn't come up as much, but in general it's better to have the list so you can compile in support for multiple compute capabilities together.  Like if you wanted to run the same binary with reasonable performance on both Pascal and Volta parts. Otherwise you have to compile for Pascal only, and lose all the special functional unit improvements.

Reply:
Looking for how to properly specify several architectures for the `-arch` flag had me digging into the rambling discussions on the topic.
The bottom line of this quest is that I couldn’t find any mention of multiple arches passed to the `-arch` option. On the contrary, I found statements that it is in facts a shorthand for when only one arch is targeted (most common and recommended use case).
Documentation suggests that only the full syntax can be used to specify multiple architectures.

So I looked into spack, which indeed accepts several architectures to the `cuda_arch` variant:
https://github.com/spack/spack/blob/be96460ab21ebc060c11d791d914c0e204d0525b/lib/spack/spack/build_systems/cuda.py#L56
Then, this is converted to cuda_flags:
https://github.com/spack/spack/blob/be96460ab21ebc060c11d791d914c0e204d0525b/lib/spack/spack/build_systems/cuda.py#L64

My conclusion, from your comment, is that I should be using the above static function embedded in Spack.

Refs:
https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#nvcc-examples 

Notes:
I found examples of the syntax to build fatbinaries for maximun compatibility here
https://arnon.dk/matching-sm-architectures-arch-and-gencode-for-various-nvidia-cards/

Reply:
Done, needs testing.